openmc/geometry.py

```diff
@@ -0,0 +1,610 @@
+
+        return material_universes
+
+    def get_all_lattices(self):
+        """Return all lattices defined
+
+        Returns
+        -------
+        collections.OrderedDict
+            Dictionary mapping lattice IDs to :class:`openmc.Lattice` instances
+
+        """
+        lattices = OrderedDict()
+
+        for cell in self.get_all_cells().values():
+            if cell.fill_type == 'lattice':
+                if cell.fill.id not in lattices:
+                    lattices[cell.fill.id] = cell.fill
+
+        return lattices
+
+    def get_all_surfaces(self):
+        """
+        Return all surfaces used in the geometry
+
+        Returns
+        -------
+        collections.OrderedDict
+            Dictionary mapping surface IDs to :class:`openmc.Surface` instances
+
+        """
+        surfaces = OrderedDict()
+
+        for cell in self.get_all_cells().values():
+            surfaces = cell.region.get_surfaces(surfaces)
+        return surfaces
+
+    def get_materials_by_name(self, name, case_sensitive=False, matching=False):
+        """Return a list of materials with matching names.
+
+        Parameters
+        ----------
+        name : str
+            The name to match
+        case_sensitive : bool
+            Whether to distinguish upper and lower case letters in each
+            material's name (default is False)
+        matching : bool
+            Whether the names must match completely (default is False)
+
+        Returns
+        -------
+        list of openmc.Material
+            Materials matching the queried name
+
+        """
+
+        if not case_sensitive:
+            name = name.lower()
+
+        all_materials = self.get_all_materials().values()
+        materials = set()
+
+        for material in all_materials:
+            material_name = material.name
+            if not case_sensitive:
+                material_name = material_name.lower()
+
+            if material_name == name:
+                materials.add(material)
+            elif not matching and name in material_name:
+                materials.add(material)
+
+        return sorted(materials, key=lambda x: x.id)
+
+    def get_cells_by_name(self, name, case_sensitive=False, matching=False):
+        """Return a list of cells with matching names.
+
+        Parameters
+        ----------
+        name : str
+            The name to search match
+        case_sensitive : bool
+            Whether to distinguish upper and lower case letters in each
+            cell's name (default is False)
+        matching : bool
+            Whether the names must match completely (default is False)
+
+        Returns
+        -------
+        list of openmc.Cell
+            Cells matching the queried name
+
+        """
+
+        if not case_sensitive:
+            name = name.lower()
+
+        all_cells = self.get_all_cells().values()
+        cells = set()
+
+        for cell in all_cells:
+            cell_name = cell.name
```

Comment:
This can be simplified to `cell_name = cell.name if case_sensitive else cell.name.lower()`